src/panic.rs

```diff
@@ -24,7 +24,7 @@ impl PanicException {
         if let Some(string) = payload.downcast_ref::<String>() {
             Self::new_err((string.clone(),))
         } else if let Some(s) = payload.downcast_ref::<&str>() {
-            Self::new_err((s.to_string(),))
+            Self::new_err(((*s).to_string(),))
```

Comment:
I think a better solution here would be to use [`payload.downcast::<&str>`](https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast). I think [`downcast_ref` introduces the extra reference](https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast_ref) clippy is complaining about.